File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        lines = matrix_string.split("\n")
        self.rows = []
    
        for line in lines:
            row = [int(num) for num in line.split()]
            self.rows.append(row)
            
    def row(self, index):
        
          return list(self.rows[index - 1])

    def column(self, index):
        columns = list(zip(*self.rows))
        return list(columns[index - 1])

File: solutions/javascript/matrix/1/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(givenString) {
    this.myMatrix = givenString.split('\n').map((row) => row.split(' ').map(Number));
  }

  get rows() {
    return this.myMatrix;
  }

  get columns() {
    return this.myMatrix[0].map((_, colIndex) =>
      this.myMatrix.map((row) => row[colIndex])
    );
  }
}
